renderdoc/driver/ihv/amd/amd_counters.cpp

```diff
@@ -26,80 +26,24 @@
-typedef GPA_Status(__stdcall *PFN_GPA_GET_PASS_COUNT)(gpa_uint32 *pNumPasses);
-typedef GPA_Status(__stdcall *PFN_GPA_BEGIN_SESSION)(gpa_uint32 *pSessionID);
-typedef GPA_Status(__stdcall *PFN_GPA_END_SESSION)();
-typedef GPA_Status(__stdcall *PFN_GPA_BEGIN_PASS)();
-typedef GPA_Status(__stdcall *PFN_GPA_END_PASS)();
-typedef GPA_Status(__stdcall *PFN_GPA_BEGIN_SAMPLE)(gpa_uint32 sampleID);
-typedef GPA_Status(__stdcall *PFN_GPA_END_SAMPLE)();
-typedef GPA_Status(__stdcall *PFN_GPA_IS_SESSION_READY)(bool *pReadyResult, gpa_uint32 sessionID);
-typedef GPA_Status(__stdcall *PFN_GPA_IS_SAMPLE_READY)(bool *pReadyResult, gpa_uint32 sessionID,
-                                                       gpa_uint32 sampleID);
-typedef GPA_Status(__stdcall *PFN_GPA_GET_SAMPLE_UINT32)(gpa_uint32 sessionID, gpa_uint32 sampleID,
-                                                         gpa_uint32 counterID, gpa_uint32 *pResult);
-typedef GPA_Status(__stdcall *PFN_GPA_GET_SAMPLE_UINT64)(gpa_uint32 sessionID, gpa_uint32 sampleID,
-                                                         gpa_uint32 counterID, gpa_uint64 *pResult);
-typedef GPA_Status(__stdcall *PFN_GPA_GET_SAMPLE_FLOAT32)(gpa_uint32 sessionID, gpa_uint32 sampleID,
-                                                          gpa_uint32 counterID, gpa_float32 *pResult);
-typedef GPA_Status(__stdcall *PFN_GPA_GET_SAMPLE_FLOAT64)(gpa_uint32 sessionID, gpa_uint32 sampleID,
-                                                          gpa_uint32 counterID, gpa_float64 *pResult);
-typedef GPA_Status(__stdcall *PFN_GPA_CLOSE_CONTEXT)();
-typedef GPA_Status(__stdcall *PFN_GPA_DESTROY)();
-
 class GPUPerfAPI
 {
 public:
-  PFN_GPA_INITIALIZE init;
-  PFN_GPA_OPENCONTEXT openContext;
-  PFN_GPA_GET_NUM_COUNTERS getNumCounters;
-  PFN_GPA_GET_COUNTER_NAME getCounterName;
-  PFN_GPA_GET_COUNTER_INDEX getCounterIndex;
-  PFN_GPA_GET_COUNTER_DESCRIPTION getCounterDescription;
-  PFN_GPA_GET_COUNTER_DATA_TYPE getCounterDataType;
-  PFN_GPA_GET_COUNTER_USAGE_TYPE getCounterUsageType;
-  PFN_GPA_ENABLE_COUNTER enableCounter;
-  PFN_GPA_ENABLE_COUNTER_STR enableCounterStr;
-  PFN_GPA_ENABLE_ALL_COUNTERS enableAllCounters;
-  PFN_GPA_DISABLE_COUNTER disableCounter;
-  PFN_GPA_DISABLE_COUNTER_STR disableCounterStr;
-  PFN_GPA_DISABLE_ALL_COUNTERS disableAllCounters;
-  PFN_GPA_GET_PASS_COUNT getPassCount;
-  PFN_GPA_BEGIN_SESSION beginSession;
-  PFN_GPA_END_SESSION endSession;
-  PFN_GPA_BEGIN_PASS beginPass;
-  PFN_GPA_END_PASS endPass;
-  PFN_GPA_BEGIN_SAMPLE beginSample;
-  PFN_GPA_END_SAMPLE endSample;
-  PFN_GPA_IS_SESSION_READY isSessionReady;
-  PFN_GPA_IS_SAMPLE_READY isSampleReady;
-  PFN_GPA_GET_SAMPLE_UINT32 getSampleUInt32;
-  PFN_GPA_GET_SAMPLE_UINT64 getSampleUInt64;
-  PFN_GPA_GET_SAMPLE_FLOAT32 getSampleFloat32;
-  PFN_GPA_GET_SAMPLE_FLOAT64 getSampleFloat64;
-  PFN_GPA_CLOSE_CONTEXT closeContext;
-  PFN_GPA_DESTROY destroy;
+  GPAApi *api;
+  GPA_GetFuncTablePtrType getFuncTable;
+
+  GPUPerfAPI()
```

Comment:
Does this class gain us anything really anymore? The `getFuncTable` member is only used during init so it could just be a local variable, and we're already manually managing the lifetime of `GPUPerfAPI m_pGPUPerfAPI` - could we just remove a layer of indirection and make `m_pGPUPerfAPI` a `GPAApi *` directly?